proposals/1412/client_proxy.md

```diff
@@ -0,0 +1,161 @@
+- Not responsive backend due to an outage or a network issue
+- Long latency of the API that adds extra delay to the processing
+
+The first scenario will likely cause the data loss which we would like to avoid as much as possible. The second one is less critical and could be mitigated by implementing asynchronous integrations, but it is very hard to do so for every single integrations.
+
+# Proposal
+In this proposal, I suggest implementing OpenLineage Proxy that is lightweight, fast, and can be run as a sidecar along with the processing engine. It helps reducing the latency for clients when the configured backend is in a different network, different a region or in a different cloud. Even when it is in the same network, this proxy will likely to provide a lower latency because it only performs the minimal work such as event validation aginst the spec, while the actual backend such as [Marquez](https://marquezproject.ai/) performs its own business logics. It also helps dealing with the backend outage, by keeping the lineage events inside its own local buffer until the backend is back available. This is of course not an ultimate solution for the problems above and there are a few other possible solutions available cloud based message broker that can act as a proxy, but it could provide an option to the users who want to deal with the problems with minimal additional resource usages while keeping the ability to handle invalid events.
+
+# Requirements
+- Proxy must implement the HTTP POST API which receives an OpenLineage JSON event according to the [OpenLineage spec](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.yml).
+- Proxy must be able to validate the OpenLineage events against the spec including standard facets.
+    - Proxy must return 200 if the event is compliant to the spec regardless of the actual OpenLineage backend service availability.
+    - Proxy must pass the event to configured *FailedEventHandler* if the event is non-compliant to the spec.
+        - Proxy should return 400 for this.
+- Proxy must buffer the OpenLineage events locally and send them to the backend when possible.
+    - If the backend is available, the proxy should send the events as soon as possible
+    - If the backend is unavailable for any reasons, proxy must keep events locally until it hits the buffer limit, and send them to the backend later when it’s available.
+        - Buffer is limited by following parameters, whatever hits the limit first
+            - T: hours to keep events to the buffer
+            - B: bytes of the messages to keep events to the buffer
+        - Buffer should be flushed when it hits the limit in order to keep the local storage available.
+- Proxy must guarantee the order of the messages to be sent to the backend as the order it is received.
+
+## Lightweight Proxy
```

Comment:
Would it help to elaborate on the difference between the proxy `backend` vs the `client`? Clearly, this docs outlines the `client` design and usage, but maybe a section on when/why a user should use proxy `backend` would be helpful.